File: lib/classes/totp_widget_data.dart
import 'package:authenticator/classes/totp_data.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'totp_widget_data.freezed.dart';
part 'totp_widget_data.g.dart';

@freezed
class TotpWidgetData with _$TotpWidgetData {
  const factory TotpWidgetData({
    required List<TotpData> data,
    required int totalAccounts,
    required String bgColor,
    required String textColor,
    required bool isUpdateFlow,
  }) = _TotpWidgetData;

  factory TotpWidgetData.fromJson(Map<String, dynamic> json) =>
      _$TotpWidgetDataFromJson(json);
}

File: android/app/src/main/kotlin/com/mkndn/authenticator/widget/TotpWidgetData.kt
package com.mkndn.authenticator.widget

import android.graphics.Color

@kotlinx.serialization.Serializable
data class TotpWidgetData(
    val data: ArrayList<TotpData> = arrayListOf(),
    val totalAccounts: Int = 0,
    val bgColor: String? = null,
    val textColor: String? = null,
    val isUpdateFlow: Boolean? = null,
)

@kotlinx.serialization.Serializable
data class TotpData(val id: String,
                    val algorithm: Algorithm,
                    val period: Int,
                    val digits: Int,
                    val label: String,
                    val issuer: String,
                    val secret: String,
                    var code: String? = null,
                    var progress: Int = period)
